r/R/duckdb.R

```diff
@@ -155,6 +154,5 @@ to_arrow <- function(.data) {
   # Run the query
   res <- DBI::dbSendQuery(dbplyr::remote_con(.data), dbplyr::remote_query(.data), arrow = TRUE)
 
-  # TODO: we shouldn't need $read_table(), but we get segfaults when we do.
-  arrow_dplyr_query(duckdb::duckdb_fetch_record_batch(res)$read_table())
+  duckdb::duckdb_fetch_record_batch(res)
```

Comment:
If you keep the `arrow_dplyr_query()` around this here then you don't need to add all of the RecordBatchReader methods

Reply:
Right, that's what I had originally. But I found that (especially for testing) being able to return a RecordBatchReader gave more flexibility (and to be able to start a dplyr query with a RecordBatchReader without needing to reach in and wrap `arrow_dplyr_query()` was also nice).

Besides the extra characters, I don't see an issue with allowing RBRs to be used like this — but LMK if I'm wrong about that

Reply:
It's fine to add the RBR methods, though I'm not sure how useful they are in practice outside of this. The other concern is: suppose someone ends their pipeline with `to_arrow()` and they get an RBR back. RBR doesn't have many useful methods defined for it (`print()`, `names()`, etc.) yet, so it will be awkward to work with if you don't know that you can `dplyr` things on it. Maybe those methods should be added. Seems like scope creep here though.

Reply:
FTR `as_arrow_dplyr_query()` exists as `as_adq()` to avoid the mouthfulness. But it is currently neither generic nor exported.

Reply:
I think the "is `to_arrow()` too general" is a separate question from what I was raising, but it is a valid concern.